dali/pipeline/operator/builtin/external_source.h

```diff
@@ -469,10 +485,24 @@ class ExternalSource : public Operator<Backend>, virtual public BatchSizeProvide
     // pass anything as it is ignored.
     std::list<uptr_tl_type> tl_elm;
     std::list<uptr_tl_type> tv_elm;
-    if (no_copy_) {
-      ShareUserData(batch, stream, use_copy_kernel);
+
+    bool actual_no_copy = no_copy_;
+    switch (set_data_config.no_copy_mode) {
+      case DALIExtNoCopyMode::FORCE_COPY:
+        actual_no_copy = false;
+        break;
+      case DALIExtNoCopyMode::FORCE_NO_COPY:
+        actual_no_copy = true;
+        break;
+      default:
+        actual_no_copy = no_copy_;
+        break;
```

Comment:
```suggestion
      default:
        break;
```

Reply:
I think I prefer to write it twice rather than miss it once.